src/constructs/autoscaling/asg.test.ts

```diff
@@ -301,4 +303,120 @@ describe("The GuAutoScalingGroup", () => {
       },
     });
   });
+
+  test("has an instance role created by default with AssumeRole permissions", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      app: "TestApp",
+      userData: "SomeUserData",
+      vpc,
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      AssumeRolePolicyDocument: {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: { Service: { "Fn::Join": ["", ["ec2.", { Ref: "AWS::URLSuffix" }]] } },
+          },
+        ],
+      },
+      Tags: alphabeticalTags([
+        { Key: "App", Value: "TestApp" },
+        { Key: "gu:cdk:version", Value: "TEST" },
+        { Key: "Stack", Value: "test-stack" },
+        { Key: "Stage", Value: { Ref: "Stage" } },
+      ]),
+    });
+  });
+
+  test("passing in an instance role overrides the default", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      app: "TestApp",
+      userData: "UserData",
+      vpc,
+      role: new GuInstanceRole(stack, {
+        app: "TestApp",
+        additionalPolicies: [
+          new GuAllowPolicy(stack, "SomePolicy", {
+            actions: ["some:Action"],
+            resources: ["some:Resource"],
+          }),
+        ],
+      }),
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      AssumeRolePolicyDocument: {
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: { Service: { "Fn::Join": ["", ["ec2.", { Ref: "AWS::URLSuffix" }]] } },
+          },
+        ],
+        Version: "2012-10-17",
+      },
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Policy", {
+      PolicyDocument: {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Action: "some:Action",
+            Effect: "Allow",
+            Resource: "some:Resource",
+          },
+        ],
+      },
+    });
+  });
+
+  test("has sensible default scaling capacities per stage based on minimum capacity", () => {
```

Comment:
This test seems to overlap with an existing test (`adds the correct mappings when provided with minimal capacity config`) - perhaps we can combine them?

Reply:
True. I guess the difference with the existing test is that the first one [provides specific defaults](https://github.com/guardian/cdk/blob/053794f87cccc9f396c10554f3634507ae3653e5/src/constructs/autoscaling/asg.test.ts#L34), whereas this one doesn't? Would be happy to combine it into something sensible though.

Reply:
I think we are currently testing 3 scenarios:

1. Using default capacity values gives you correct mappings & references (new test)
2. Using custom capacity values gives you correct mappings & references (new test)
3. Using custom capacity values that happen to be identical to the default values gives you correct mappings (existing test)

If 1 and 2 are working, I think it's highly unlikely that 3 will be broken, so I'd probably remove it for brevity.

Reply:
Yeah, that makes sense, thanks for explaining it! I've removed the existing test as it is superfluous.